dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -74,261 +67,222 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
+		}
+	}
+
+	delete friendRes;
+	friendRes = nullptr;
+
+	delete friendQuery;
+	friendQuery = nullptr;
 }
 
-std::string PropertyEntranceComponent::BuildQuery(Entity* entity, int32_t sortMethod, Character* character, std::string customQuery, bool wantLimits) {
-	std::string base;
-	if (customQuery == "") {
-		base = baseQueryForProperties;
+std::vector<uint32_t> PropertyEntranceComponent::GetPropertyIDsBasedOnParams(const std::string& searchText, uint32_t sortMethod, Entity* requestor) {
+	std::string query = "SELECT id, owner_id, last_updated, reputation FROM properties WHERE zone_id = ? AND (description LIKE ? OR name LIKE ? OR name LIKE ?) AND privacy_option >= ? ";
+
+	if (sortMethod == SORT_TYPE_REPUTATION) {
+		query += "ORDER BY reputation DESC;";
 	} else {
-		base = customQuery;
+		query += "ORDER BY last_updated DESC;";
 	}
-	std::string orderBy = "";
-	if (sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS) {
-		std::string friendsList = " AND p.owner_id IN (";
 
-		auto friendsListQuery = Database::CreatePreppedStmt("SELECT * FROM (SELECT CASE WHEN player_id = ? THEN friend_id WHEN friend_id = ? THEN player_id END AS requested_player FROM friends ) AS fr WHERE requested_player IS NOT NULL ORDER BY requested_player DESC;");
+	auto propertyLookup = Database::CreatePreppedStmt(query);
 
-		friendsListQuery->setUInt(1, character->GetID());
-		friendsListQuery->setUInt(2, character->GetID());
+	const auto searchString = "%" + searchText + "%";
+	propertyLookup->setUInt(1, this->m_MapID);
+	propertyLookup->setString(2, searchString.c_str());
+	propertyLookup->setString(3, searchString.c_str());
+	propertyLookup->setString(4, searchString.c_str());
+	propertyLookup->setInt(5, sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS ? (uint32_t)PropertyPrivacyOption::Friends : (uint32_t)PropertyPrivacyOption::Public);
+
+	auto propertyIdRes = propertyLookup->executeQuery();
 
-		auto friendsListQueryResult = friendsListQuery->executeQuery();
+	std::vector<uint32_t> propertyIds{};
 
-		while (friendsListQueryResult->next()) {
-			auto playerIDToConvert = friendsListQueryResult->getInt(1);
-			friendsList = friendsList + std::to_string(playerIDToConvert) + ",";
+	while (propertyIdRes->next()) {
+		if (sortMethod == SORT_TYPE_FRIENDS || sortMethod == SORT_TYPE_FEATURED) {
+			if (m_UserFriendMap[(uint32_t)requestor->GetObjectID()].find(propertyIdRes->getUInt(2)) == m_UserFriendMap[(uint32_t)requestor->GetObjectID()].end()) {
+				continue;
+			}
 		}
-		// Replace trailing comma with the closing parenthesis.
-		if (friendsList.at(friendsList.size() - 1) == ',') friendsList.erase(friendsList.size() - 1, 1);
-		friendsList += ") ";
 
-		// If we have no friends then use a -1 for the query.
-		if (friendsList.find("()") != std::string::npos) friendsList = " AND p.owner_id IN (-1) ";
+		propertyIds.push_back(propertyIdRes->getUInt(1));
+	}
 
-		orderBy += friendsList + "ORDER BY ci.name ASC ";
+	delete propertyIdRes;
```

Comment:
nit: Unique_ptrs on all queries and results to reduce clutter sloc and improve memory safety and not need to worry about doing deletes in other return paths as you do below